### DISCUSSION TITLE
Where can I download example panels and Lua Scripts?

### DISCUSSION THREAD

**Discussion by thomas0906** [RomanKubiak/ctrlr#311] [Q&A]
Hi
I am new.
What a great software!
I am a software developer and I know Lua.
I need to create a panel for Yamaha Genos Keyboard.
I am working with Ableton Live Standard DAW.
But I need some examples on how to build and to use all the possibilities of Ctrlr.

Where can I find downloads?
Registering the old forum seems not possible.

Regards
Thomas

**Comment by Ted-juh** [comment]
You can download panels which are open source, if you are lucky, from the downloads section of the CTRLR site. There are demo files in the CTRLR folder. And the Juce site is a welcome addition to see all the classes that you "might"  need "if" they are fully supported.

Welcome to the club and good luck. The Genos does look like a quite extended workstation. Start small and extend the panel once your knowledge of Juce/ CTRLR expands.

**Reply by thomas0906** [reply]
Hi thanx, but I can not download without a login. And I get no login password.

**Reply by Ted-juh** [reply]
Ah ok, I see. I thought you only had to log in to place a comment in the downloads section. Apparently not. A lot of panels use the blue "download" button, those panels are hosted on WordPress and need a login. But a quick look shows that a few panels are hosted on Sourceforge or other sites and those can be downloaded "without" a login. But if there is a specific panel that you want that is hosted on the Ctrlr site that you can't download, just ask. I can email it to you or put it on my own Github page for you to download.

**Reply by damiensellier** [reply]
Hi @thomas0906, welcome aboard. 
When I started I did stumble uppon those 2 youtube videos that really helped me understand how Ctrlr works:
https://www.youtube.com/watch?v=lszMkl3Jp1k
https://www.youtube.com/watch?v=yf3HSyfbwuk

Then I read Dominique's pdf guide:
https://ctrlr.org/wp-content/uploads/2016/03/Ctrlr-Step-by-Step-v13.pdf

And you can download demo pannels for some of the app features here : 
https://github.com/RomanKubiak/Panels/tree/1aaeb12ad9cd96a16cddb37fd2a2c8ca98f87a01/DEMO

Ctrlr relies on JUCE the famous audio framework, and LUA for coding instead of C++. So you'll need to read a lot the JUCE documentation. Not all classes are implemented in Ctrlr.

We used to have the forum on ctrlr.org, it's still available and is full of resources.
But most of all, we have this github discussion where people are really helpful if you meet a difficulty.

Good luck with your project ;)

**Reply by Godlike-Productions** [reply]
https://ctrlr.org/panels/ is the place to get them.  It works for me, even without a login. Panels that are hosted at ctrlr can be downloaded eg https://ctrlr.org/mixer-panel-for-the-roland-jupiter-x-m/

Some panels are hosted by third parties, such as mine and Goodweathers, and the links seem to work (though not all of those have bpanelz available for you to interrogate.

Great starting points are the Miniak - https://ctrlr.org/akai-miniak-alesis-micron/ panel and DSI Pro 2 - https://ctrlr.org/dsi-pro2-editor-and-librarian/

Having said that, it might be an idea for us to start uploading panels here - https://github.com/RomanKubiak/Panels/tree/1aaeb12ad9cd96a16cddb37fd2a2c8ca98f87a01

**Reply by dobo365** [reply]
Thx guys for referencing my Step by Step guide and Pro-2 panel!

**Reply by Ted-juh** [reply]
Almost don't dare to ask but are you making progress with the step by step guide version 2.0? Do you need any help? Do you need contributions for certain areas? Anything to get this beast of a handbook see they light of day?

**Reply by Godlike-Productions** [reply]
@dobo365 , are you happy for me to upload a copy of the guide to  https://godlike.com.au/index.php?id=361
